tests/foreman/ui_airgun/test_host.py

```diff
@@ -134,6 +145,248 @@ def module_host_template(module_org, module_loc):
+    # location and dns to it
+    _, _, domain = settings.server.hostname.partition('.')
+    domain = entities.Domain().search(
+        query={u'search': u'name="{0}"'.format(domain)}
+    )
+    if len(domain) > 0:
+        domain = domain[0].read()
+        domain.location.append(module_loc)
+        domain.organization.append(module_org)
+        domain.dns = module_proxy
+        domain = domain.update(['dns', 'location', 'organization'])
+    else:
+        domain = entities.Domain(
+            dns=module_proxy,
+            location=[module_loc],
+            organization=[module_org],
+        ).create()
+    return domain
+
+
+@pytest.fixture(scope='module')
+def module_libvirt_subnet(
+        module_org, module_loc, module_libvirt_domain, module_proxy):
+    # Search if subnet is defined with given network.
+    # If so, just update its relevant fields otherwise,
+    # Create new subnet
+    network = settings.vlan_networking.subnet
+    subnet = entities.Subnet().search(
+        query={u'search': u'network={0}'.format(network)}
+    )
+    if len(subnet) > 0:
+        subnet = subnet[0].read()
+        subnet.domain.append(module_libvirt_domain)
+        subnet.location.append(module_loc)
+        subnet.organization.append(module_org)
+        subnet.dns = module_proxy
+        subnet.dhcp = module_proxy
+        subnet.ipam = 'DHCP'
+        subnet.tftp = module_proxy
+        subnet.discovery = module_proxy
+        subnet = subnet.update([
+            'domain', 'discovery', 'dhcp', 'dns', 'ipam', 'location', 'organization', 'tftp'])
+    else:
+        # Create new subnet
+        subnet = entities.Subnet(
+            network=network,
+            mask=settings.vlan_networking.netmask,
+            location=[module_loc],
+            organization=[module_org],
+            domain=[module_libvirt_domain],
+            ipam='DHCP',
+            dns=module_proxy,
+            dhcp=module_proxy,
+            tftp=module_proxy,
+            discovery=module_proxy,
+        ).create()
+    return subnet
+
+
+@pytest.fixture(scope='module')
+def module_libvirt_lce(module_org):
+    return entities.LifecycleEnvironment(organization=module_org).create()
+
+
+@pytest.fixture(scope='module')
+def module_libvirt_product(module_org):
+    return entities.Product(organization=module_org).create()
+
+
+@pytest.fixture(scope='module')
+def module_libvirt_repository(os_path, module_libvirt_product):
+    old_task_timeout = entity_mixins.TASK_TIMEOUT
+    repo = entities.Repository(
+        product=module_libvirt_product, url=os_path).create()
+    entity_mixins.TASK_TIMEOUT = 3600
+    repo.sync()
+    entity_mixins.TASK_TIMEOUT = old_task_timeout
+    return repo
+
+
+@pytest.fixture(scope='module')
+def module_libvirt_content_view(module_org, module_libvirt_repository, module_libvirt_lce):
+    # Create, Publish and promote CV
+    old_task_timeout = entity_mixins.TASK_TIMEOUT
+    entity_mixins.TASK_TIMEOUT = 3600
```

Comment:
```
old_task_timeout = entity_mixins.TASK_TIMEOUT
entity_mixins.TASK_TIMEOUT = 3600
```
set and uset timeout used multiple times, can we put that into a common function?

Reply:
Did you see that we do?:
```
timeout = new
doing some functionality
timeout =old
```
not sure how to make it generic. Do you propose to create some common function that will receive functions as a parameter. But we can have multiple statements, so lambda or similar ideology doesn't fit our needs.

Reply:
That's OK then. I'm fine with it.

Reply:
we cannot be sure that the TASK_TIEOUT is restored  in case of failure we can use
call_entity_method_with_timeout for each function as commented above or 
```python
old_task_timeout = entity_mixins.TASK_TIMEOUT
entity_mixins.TASK_TIMEOUT = 3600
try:
    ....

finally: 
     entity_mixins.TASK_TIMEOUT = old_task_timeout

```